src/transformers/models/llama/modeling_llama.py

```diff
@@ -195,39 +195,34 @@ def apply_rotary_pos_emb(q, k, cos, sin, position_ids):
     k_embed = (k * cos) + (rotate_half(k) * sin)
     return q_embed, k_embed
 
-# For PyTorch/XLA's SPMD 2D sharding
+# For PyTorch/XLA's SPMD sharding
 def init_spmd(model, config):
-    num_devices = xr.global_runtime_device_count()
     model.spmd_debug = config.spmd_debug
-    model.spmd_model_axis = config.spmd_model_axis
-    model.spmd_data_axis = num_devices // model.spmd_model_axis
-    assert model.spmd_data_axis * model.spmd_model_axis == num_devices
-    model.spmd_iota_mesh = config.spmd_iota_mesh
-
-def get_mesh(spmd_iota_mesh, ici_mesh_shape, dcn_mesh_shape=None):
-    if spmd_iota_mesh:
-        if dcn_mesh_shape is not None:
-            assert len(ici_mesh_shape) == len(dcn_mesh_shape)
-            for i in range(len(dcn_mesh_shape)):
-                ici_mesh_shape[i] *= dcn_mesh_shape[i]
-        num_devices = xr.global_runtime_device_count()
-        device_ids = torch.arange(num_devices)
-        return xs.Mesh(device_ids, ici_mesh_shape)
-    else:
-        return xs.HybridMesh(ici_mesh_shape=ici_mesh_shape, dcn_mesh_shape=dcn_mesh_shape)
+    model.spmd_mesh = config.spmd_mesh
+    model.spmd_fsdp_sharding = config.spmd_fsdp_sharding
 
 class LlamaMLP(nn.Module):
     def __init__(self, config):
         super().__init__()
         self.config = config
+        init_spmd(self, config)
+
         self.hidden_size = config.hidden_size
         self.intermediate_size = config.intermediate_size
-        self.gate_proj = nn.Linear(self.hidden_size, self.intermediate_size, bias=False)
-        self.up_proj = nn.Linear(self.hidden_size, self.intermediate_size, bias=False)
-        self.down_proj = nn.Linear(self.intermediate_size, self.hidden_size, bias=False)
+        self.gate_proj = nn.Linear(self.hidden_size, self.intermediate_size, bias=False).to('xla')
+        self.up_proj = nn.Linear(self.hidden_size, self.intermediate_size, bias=False).to('xla')
+        self.down_proj = nn.Linear(self.intermediate_size, self.hidden_size, bias=False).to('xla')
         self.act_fn = ACT2FN[config.hidden_act]
 
-        init_spmd(self, config)
+        if self.spmd_fsdp_sharding:
```

Comment:
Are we dropping tensor sharding?

Reply:
I feel like we should drop it as it's not useful anyway and then it's not compatible with the 2D shading scheme. Let me do a follow up to drop it later.